packages/@lwc/engine-core/src/framework/freeze-template.ts

```diff
@@ -103,18 +137,15 @@ function warnOnStylesheetFunctionMutation(stylesheet: StylesheetFactory) {
                 return value;
             },
             set(newValue) {
-                logError(
-                    `Dynamically setting the "${prop}" property on a stylesheet function ` +
-                        `is deprecated and may be removed in a future version of LWC.`
-                );
+                reportStylesheetViolation(prop);
                 value = newValue;
             },
         });
     }
 }
 
 // Warn on either array or stylesheet (function) mutation, in a deeply-nested array
-function warnOnStylesheetsMutation(stylesheets: TemplateStylesheetFactories) {
+function trackStylesheetsMutation(stylesheets: TemplateStylesheetFactories) {
```

Comment:
minor: is this function's for both template and stylesheet mutation warning? Should we rename it to something like `trackTemplateAndStylesheetMutation` to be more clear?

Reply:
This particular function actually only tracks the `stylesheets` array. `trackMutations` does both.

Reply:
Right. Sorry, I think I got confused as `warnOnArrayMutation` calls `reportTemplateViolation` but it is actually on `stylesheets` array of the template.